tests/benchmarks/test_parquet.py

```diff
@@ -0,0 +1,118 @@
+"""
+Parquet-related benchmarks.
+"""
+import uuid
+
+import dask.dataframe as dd
+import dask.datasets
+import distributed
+import fsspec
+import pandas
+import pytest
+from coiled.v2 import Cluster
+
+N_WORKERS = 15
+
+
+@pytest.fixture(scope="module")
+def parquet_cluster():
+    with Cluster(
+        f"parquet-{uuid.uuid4().hex[:8]}",
+        n_workers=N_WORKERS,
+        worker_vm_types=["m5.xlarge"],
+        scheduler_vm_types=["m5.xlarge"],
+    ) as cluster:
+        yield cluster
+
+
+@pytest.fixture(scope="module")
+def parquet_client(parquet_cluster):
+    with distributed.Client(parquet_cluster) as client:
+        parquet_cluster.scale(N_WORKERS)
+        client.wait_for_workers(N_WORKERS)
+        client.restart()
+        yield client
+
+
+def test_read_spark_generated_data(parquet_client):
+    """
+    Read a ~15 GB subset of a ~800 GB spark-generated
+    open dataset on AWS.
+
+    The dataset was copied from AWS open data on 2022-05-25
+    https://registry.opendata.aws/1000-genomes-data-lakehouse-ready/
+    Citation: https://www.nature.com/articles/s41467-018-08148-z
+    """
+    ddf = dd.read_parquet(
+        "s3://coiled-runtime-ci/thousandgenomes_dragen/var_partby_samples/NA21**.parquet",
+        engine="pyarrow",
+        index="sample_id",
+    )
+    ddf.groupby(ddf.index).first().compute()
+
+
+def test_read_hive_partitioned_data(parquet_client):
+    """
+    Read a dataset partitioned by year and quarter.
+
+    The dataset was copied from AWS open data on 2022-05-25
+    https://registry.opendata.aws/speedtest-global-performance/
+    """
+    ddf = dd.read_parquet(
+        "s3://coiled-runtime-ci/ookla-open-data/type=fixed/**.parquet",
+        engine="pyarrow",
+    )
+
+    # The data is already partitioned by year and quarter, but it doesn't
+    # fit Dask's partitioning scheme well since we don't support multiindexes.
+    # This is a lot of work to set the index for something that is already
+    # partitioned! We also go around dask's set_index, which does a huge amount
+    # of extra work, and even takes down instances.
+    def get_period(df):
+        return df.set_index(
+            pandas.PeriodIndex(
+                df.year.astype(str) + "Q" + df.quarter.astype(str),
+                freq="Q",
+                name="period",
+            )
+        )
```

Comment:
This step isn't really necessary to correctly load the dataset (I could delete this and the test would still pass) -- it's really more illustrative of a hoop that we need to jump through to get a *useful* set of divisions.

If we could handle multi-indexes or similar, then this would go away.